blackduck/Client.py

```diff
@@ -102,6 +103,94 @@ def __init__(
         self.base_url = base_url
         self.session = session or HubSession(base_url, timeout, retries, verify)
         self.session.auth = (auth or BearerAuth)(session=self.session, token=token, username=username, password=password)
+        self.root_resources_dict = None
+
+    def list_resources(self, parent=None):
+        """List resources that can be fetched
+
+        Args:
+            self:
+            parent (dict/json): resource object from prior get_resource invocations.
+                                Defaults to None (for root /api/ base).
+
+        Returns:
+            dict(str -> str): of public resource names to urls
+        """
+        if parent is not None and not isinstance(parent, dict):
+            raise TypeError("parent parameter must be a dict if not None")
+        if not parent:
+            # the root resources are in a different format (name -> href)
+            # compared to (rel, href) pairs in _meta.links
+            if self.root_resources_dict is None:
+                # cache root resources for efficiency
+                resp_json = self.session.get("/api/").json()
+                del resp_json['_meta']
+                self.root_resources_dict = resp_json
+            return self.root_resources_dict
+        else:
+            key = '_hub_rest_api_python_resources_dict'
+            if key not in parent:
+                obj = safe_get(parent, '_meta', 'links')
+                try:
+                    rel_href_pairs = iter(obj)
+                except TypeError:
+                    logger.error("not iterable bd_object:")
+                    pprint(obj)
+                    raise
+                resources_dict = {}
+                for res in rel_href_pairs:
+                    resources_dict[res['rel']] = res['href']
+                parent[key] = resources_dict  # cache for future use
+            return parent[key]
+
+    def get_resource(self, name, parent=None, items=True, **kwargs):
```

Comment:
@AR-Calder I removed the public=True argument. I think if the developer is going to be hitting non-public areas of the API, they should use Client.session.get() directly and can sling together URLs manually before sending the request. Let me know though if you think it is critical and I can work it back in.